lte/gateway/c/core/oai/common/common_defs.h

```diff
@@ -51,11 +51,21 @@
 #define COUNT_OF(x) \
   ((sizeof(x) / sizeof(0 [x])) / ((size_t)(!(sizeof(x) % sizeof(0 [x])))))
 
+#ifndef __cplusplus
 #define PARENT_STRUCT(cOnTaiNeD, TyPe, MeMBeR)                    \
   ({                                                              \
     const typeof(((TyPe*)0)->MeMBeR)* __MemBeR_ptr = (cOnTaiNeD); \
     (TyPe*)((char*)__MemBeR_ptr - OFFSET_OF(TyPe, MeMBeR));       \
   })
+#endif
+
+#ifdef __cplusplus
+#define PARENT_STRUCT(cOnTaiNeD, TyPe, MeMBeR)                    \
+  ({                                                              \
+    const decltype(((TyPe*)0)->MeMBeR)* __MemBeR_ptr = (cOnTaiNeD); \
+    (TyPe*)((char*)__MemBeR_ptr - OFFSET_OF(TyPe, MeMBeR));       \
```

Comment:
```suggestion
    (TyPe*)((char*)__MeMBeR_ptr - OFFSET_OF(TyPe, MeMBeR));       \
```

Reply:
How in the world is this compiling... Now I question things.

Reply:
AAH the UI is what is confusing me here. This is a SUGGESTION not a snippet of my diff (which appears it would not compile if applied). So you are suggesting that your pattern match estimate of the naming convention should result in `__MeMBeR_ptr` (based upon their use of `MemBeR` probably? That seems reasonable to assume. I would fall back to "I was trying to not change things other than the critical change in this PR". How about I post a follow-up PR to change all the naming of this macro?

Reply:
:crossed_fingers:

Reply:
Once this PR is merged I'll float [this renaming PR](https://github.com/electronjoe/magma/commit/415bb8e81c0e90f60dbbba7b37dde53f7a7d738a)